### ISSUE TITLE
[ FEATURE REQUEST ] New sources (See description)

### ISSUE THREAD

**Issue by ohlookitsderpy** [Yucked/Frostbyte#4]
### `Feature Description:`

Could you add the following sources please? (Not sure if some are currently implemented/planned but just listing ideas)

Most Wanted:
* niconico (https://www.nicovideo.jp)
* Newgrounds (https://www.newgrounds.com)
* Twitter (https://twitter.com)

Others:
* bilibili (https://www.bilibili.com)
* Dailymotion (https://www.dailymotion.com)
* hearthis.at (https://hearthis.at)

Thanks!

**Comment by Yucked** [comment]
Hey those most wanted sources seems like they would fall under HTTP source where you send a direct link to video/audio. I'm not sure about the others but if anything they will be added once the basic system is up and running.

**Comment by ohlookitsderpy** [comment]
They do, but as with YouTube it is time-consuming to manually get the URLs

**Comment by Yucked** [comment]
Makes sense. That gives me an idea of an addon feature thing where you load sources on the go.

**Comment by ohlookitsderpy** [comment]
Sounds interesting!

**Comment by TheMulti0** [comment]
I would like to see Radio Garden (https://radio.garden) as a source in Frostbyte

**Comment by Earu** [comment]
> 
> 
> I would like to see Radio Garden (https://radio.garden) as a source in Frostbyte

Im pretty sure radio.garden does not have a public API (it does have one tho). They seem very protective of their radio streams. Here is something that might help you if you decide to implement it:

- radio garden url: https://radio.garden/listen/array-107/c7A3kGrB
- actual media stream: http://edge1-b.exa.live365.net/a32108?listening-from-radio-garden=1562057579025

There must some kind of links between these two.

**Comment by TheMulti0** [comment]
> > I would like to see Radio Garden (https://radio.garden) as a source in Frostbyte
> 
> Im pretty sure radio.garden does not have a public API (it does have one tho). They seem very protective of their radio streams. Here is something that might help you if you decide to implement it:
> 
> * radio garden url: https://radio.garden/listen/array-107/c7A3kGrB
> * actual media stream: http://edge1-b.exa.live365.net/a32108?listening-from-radio-garden=1562057579025
> 
> There must some kind of links between these two.

They have an API which is partially publicly open, the fact that it does not require authorization does not mean it is private.
a GET request to `https://radio.garden/api/content/places` will give you a JSON with all of the possible radio stations with their IDs.
Then https://radio.garden/listen/array-107/`ID` will give you the website link, and I'm sure getting the HTTP stream out of the website should not require more than 2 or 3 redirect requests, I can look into it and search for some kind of connection between the audio stream and information from the API..
